### PR TITLE
#404 Historical Cases chart on facility page 

### LINKED ISSUE #404: Viewing historical case data on the facility detail page
**Is your feature request related to a problem? Please describe.**
Currently, users have no way of viewing the historical case and population data they have entered for a facility. This causes problems because it is difficult to debug a faulty R(t) and hard to know where they are missing data. We also want to encourage them to give us a complete set of data so we can give them the most accurate projection.

**Describe the solution you'd like**
Because they add cases on a per-facility level, we want to add a chart that allows them to view the case data they have inputted historically and easily identify where they are missing data.

image

image

Components

1. Bar chart by default shows the most recent 30 days. 
2. Clicking “Previous 15 days” scrolls the chart to be able to show the previous 15 days of data (if there is an easier chart library to handle scrolling back in time, feel free to use that)
3. Each bar is a click target that opens the “add historical data” modal with the date of the bar pre-selected
4. When there’s no data, the bar is red and stretches the full y-axis length of the chart (On click, opens the “Add historical data” modal with the date pre-selected to the date of the bar. On hover, color gets darker to show that the bar is a click target. Clicking on the hover text should also open the modal)
5. When there is data for a day, the bar is green and has a height of the # of confirmed cases (On click, opens the “Add historical data” modal with the date pre-selected to the date of the bar, with data pre-populated from what the user previously entered). On hover, shows the date, case count, and a CTA to click to change the case count. Click on the hover text should also open the modal)
6. P2: If a case count is 0 or less than 50% between the first y-axis line and 0, display the count above the bar
7. Button at the top right “+ Add Historical Data” to open the “add historical data” modal with yesterday’s date pre-selected (or the default we are currently using)

Mocks: https://www.figma.com/file/6z1QQW84a59Ul3c17PgWdV/%F0%9F%9A%80Recidiviz-CJStatus-(excalibur)?node-id=2206%3A6804

**Describe alternatives you've considered**
We considered showing a table or calendar widget that indicates historical data and data missing. A bar chart lets them see the trend over time. A line chart was also considered, but it is more challenging to interact with.

**Additional context**
This information is repeated in this doc, in case there are complications that warrant a longer discussion: https://docs.google.com/document/d/1KRCYPeC3W7KJm7flUpZKPxdhL7U0sRTfisD0cjqQYuM/edit#heading=h.xr8mpbcylhxh

Related to #405

### PR DESCRIPTION
**Pull request by daschi**
## Description of the change

I'm opening this PR as a draft to keep everyone updated on the progress and what is left to do. If anyone is interested in working on this too I think some of these tasks could be worked on separately! We could use this PR to track progress, and if someone wants to work on one of the todos, I was thinking you could just edit this and put your name next to it? Totally open to other ideas, let me know what you think. 

What I'm working on next: 
- [x] Use the sum of cases for incarcerated and staff counts
- [x] Add ability to click to `next 15 days` and `previous 15 days`
- [x] (Daniela)Add text in tooltip that says "Click to update"
- [x] (Daniela) Add an onClick to the bars to open the Add Cases modal with auto-populated date from the clicked bar and auto-populated cases data if it exists
- [x] (Daniela) Format the font styles in the tooltip, pull the tooltip out into a generic component that can be shared

**TODO:** 

- [x] [Soumya #459 ]Change the bar's color to a darker color on hover to signify it as a click target
- [x] [Soumya #459 ] Add the chart header with the Cases title and a button that opens the Add Cases modal with yesterday's date pre-selected (or what the current default date is)
- [x] [Daniela] Format the style of the y and x axis font & color
- [x] [Daniela] Chart width becomes too wide when screen size becomes smaller

**TBD**
- [ ] **P2/might not implement**: Display month's name under the x-axis at certain points (?)
- [ ]  **TBD**: If case count is 0 or less than 50% of the y-axis, display the count above the bar
- [ ] **TBD w/Serena first** Display empty bar chart when adding a new facility and allow user to add historical data in the add cases modal for a new facility

Some possible blockers:

- ~I still need to investigate whether it'll be possible to position the tooltip on hover so that a user can click on a link inside it~
- ~I had some trouble figuring out how to display the x axis with the two rows of labels (i.e. the date and the month underneath the 1), so for now I formatted the x-axis label as `MM/dd`~
- **Nice-to-have:** Currently the tooltip appears at the top of the bar (or at the bottom if there is no data), I'm not sure if we want to change this so that it sometimes appears in the middle of the bar?

**Note:** @sychang and I chatted about removing the red bar that is shown in the mocks, and showing no bar for now when there are 0 cases. 

Here's where it is so far visually (see linked issue for the mock):
image

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #404 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by sychang** [comment]
For the blockers:
I still need to investigate whether it'll be possible to position the tooltip on hover so that a user can click on a link inside it 
--> Let's do away with the link inside the tooltip and make it say "click to update" with no link instead for simplification

I had some trouble figuring out how to display the x axis with the two rows of labels (i.e. the date and the month underneath the 1), so for now I formatted the x-axis label as MM/dd 
--> MM/dd is totally fine, let's do away with the two rows of labels for simplification

Currently the tooltip appears at the top of the bar (or at the bottom if there is no data), I'm not sure if we want to change this so that it sometimes appears in the middle of the bar?
--> I would say this is nice-to-have but definitely not a launch-blocker

Now that we are no longer using the red bars to show missing data, I would change this one slightly: "P2: If case count is 0 or less than 50% of the y-axis, display the count above the bar" 
--> The minimum would be visually distinguishing "0 cases" from "missing data" (which would both currently appear to be empty bars). Whoever ends up taking that component, let's chat about through a few options that are easiest. TODO: I will come back and update this comment with some options and what we decide

**Comment by daschi** [comment]
@jessex I believe this PR along with @SoumyaMahadevan's PR #459 are ready for review. Let us know if it would be easier to review them together in one PR

**Comment by jessex** [comment]
Thanks, @daschi and @SoumyaMahadevan!

My plan is to test on Soumya's PR, review Daniela's PR, then review Soumya's PR. Then assuming all is good, the easiest path is probably to merge Soumya's PR into Daniela's, and then merge Daniela's into master.

**Comment by rocktacular** [comment]
ok got through it all and think I understand what's happening. Lots of new functionality and seems fine to me. I haven't looked at Soumya's PR, but I agree with Josh that pulling hers into this one before going to master is probably easiest.

I see 1 linting issue you may want to address.

File: src/hooks/useAddCasesInputs.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+  const [inputs, setInputs] = useState<ModelInputsPopulationBrackets>(
+    defaultInputs,
+  );
+
+  const [observationDate, setObservationDate] = useState(
+    defaultObservationDate,
+  );
+
+  // maybe export this so that we don't define it twice (it is also in the AddCasesModalContent component)
```
Comment (jessex) [inline src/hooks/useAddCasesInputs.tsx:45]: Would be good, but I also don't mind filing a tech debt issue to loop back around and handle this, given how much effort has already gone into this feature and its overall importance!

File: src/hooks/useAddCasesInputs.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        setInputs(getModelInputs(observedAtVersion));
+        setObservedAtVersion(observedAtVersion);
+      } else {
+        setInputs({});
+        setObservedAtVersion(undefined);
+      }
+    }
+  }, [observedAt]);
+
+  // TODO: Handle add cases modal for new facility?
```
Comment (jessex) [inline src/hooks/useAddCasesInputs.tsx:69]: This todo is irrelevant, right? The modal isn't shown when you're creating a brand new facility.

File: src/hooks/useAddCasesInputs.tsx (hunk 1 above)
Comment (daschi) [reply src/hooks/useAddCasesInputs.tsx:47]: Okay sounds good, I'll remove the note for now.

File: src/hooks/useAddCasesInputs.tsx (hunk 2 above)
Comment (daschi) [reply src/hooks/useAddCasesInputs.tsx:71]: Yes I can remove this! thanks

**Comment by daschi** [comment]
@jessex and @rocktacular thanks for the reviews! I've updated it so there's more margin on the right side of the x-axis and added the toast back in (oops). I think this is ready for another look, let me know if there's any other issues.

**Pull request merged**